### PR TITLE
fixed Toast is briefly displayed on page load

### LINKED ISSUE #2512: Toast is briefly displayed on page load
On several of our pages, the chromedash-toast element briefly appears in the bottom left corner if the page is initially loaded or reloaded (not on a SPA navigation).   The fact that the toast flashed by looks like a mistake or error, even though it does not prevent the user from accomplishing any use case.  We should prevent it because it is not intended to be initially displayed.

### PR DESCRIPTION
**Pull request by r3yc0n1c**
fixes #2512 

### Summary:
- Resolved the toast-display issue on page load in multiple pages like roadmap, myfeatures, newfeatures, and probably others.

### Tests
- [x] Tested the toast display on roadmap, myfeatures, newfeatures, etc. pages
- [x] Tested correct workflow of the toast popups like "Link copied", "Some errors occurred", etc.

### Screenshots

https://github.com/GoogleChrome/chromium-dashboard/assets/57730879/6961fbf7-d99e-4d8a-8260-e9ef3d9de299

This approach works on every page, but having a welcome toast on every page was not ideal. Previously, due to the <chromedash-toast> initialization, the toast used to appear briefly like that. Thus, I've commented out the toast initialization with the welcome text in the `spa` and `_base`.

For now, I've considered initializing the toast only where needed, thus solving this issue.

I'm open to discussing and implementing any other approach. I would appreciate a review of this! Thanks!

cc: @DanielRyanSmith @jrobbins

### CODE REVIEW

File: templates/_base.html [hunk 1]
```diff
@@ -126,7 +126,7 @@
     <chromedash-footer></chromedash-footer>
   </div>
 
-  <chromedash-toast msg="Welcome to chromestatus.com!"></chromedash-toast>
+  <!-- <chromedash-toast msg="Welcome to chromestatus.com!"></chromedash-toast> -->
```
[the remark this answers is not in the document]
Comment (r3yc0n1c) [reply templates/_base.html]: done :raised_hands:

File: templates/spa.html [hunk 2]
```diff
@@ -78,7 +78,7 @@
     bannerTime="{{banner_time}}">
   </chromedash-app>
 
-  <chromedash-toast msg="Welcome to chromestatus.com!"></chromedash-toast>
+  <!-- <chromedash-toast msg="Welcome to chromestatus.com!"></chromedash-toast> -->
```
[the remark this answers is not in the document]
Comment (r3yc0n1c) [reply templates/spa.html]: done as well

File: client-src/elements/chromedash-roadmap-page.js [hunk 3]
```diff
@@ -57,6 +57,10 @@ export class ChromedashRoadmapPage extends LitElement {
 
   firstUpdated() {
     this.handleResize();
+
+    const toastEl = document.createElement('chromedash-toast');
+    document.body.appendChild(toastEl);
+    toastEl.showMessage("Welcome to chromestatus.com!");
```
[the remark this answers is not in the document]
Comment (r3yc0n1c) [reply client-src/elements/chromedash-roadmap-page.js]: changed it as suggested :raised_hands:

**Comment by DanielRyanSmith** [comment]
This is looking like the solution we're looking for, but it seems some of the test reference files need to be regenerated, since they no longer have the `chromedash-toast` element in them. Take a look at the "Details" section of the python_tests check that has failed, and follow the [instructions on the readme](https://github.com/GoogleChrome/chromium-dashboard?tab=readme-ov-file#developing) to update the test html renderings.

E.g. for one of the tests [you'll need to uncomment this line](https://github.com/GoogleChrome/chromium-dashboard/blob/main/pages/featurelist_test.py#L152) and run `npm test`, which will regenerate the rendering, then comment it out again and make sure the test passes after. Then commit that change to the html rendering.

**Comment by r3yc0n1c** [comment]
> This is looking like the solution we're looking for, but it seems some of the test reference files need to be regenerated, since they no longer have the `chromedash-toast` element in them. Take a look at the "Details" section of the python_tests check that has failed, and follow the [instructions on the readme](https://github.com/GoogleChrome/chromium-dashboard?tab=readme-ov-file#developing) to update the test html renderings.
> 
> E.g. for one of the tests [you'll need to uncomment this line](https://github.com/GoogleChrome/chromium-dashboard/blob/main/pages/featurelist_test.py#L152) and run `npm test`, which will regenerate the rendering, then comment it out again and make sure the test passes after. Then commit that change to the html rendering.

fixed the html test pages and tested with `npm test` but this error pops up:

```sh
FAIL: test__normal (internals.approval_defs_test.FetchOwnersTest.test__normal)
We can fetch and parse an OWNERS file.  And reuse cached value.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/rey/.pyenv/versions/3.11.8/lib/python3.11/unittest/mock.py", line 1375, in patched
    return func(*newargs, **newkeywargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/rey/Documents/chromium-dashboard/internals/approval_defs_test.py", line 48, in test__normal
    mock_get.assert_called_once_with('https://example.com')
  File "/home/rey/.pyenv/versions/3.11.8/lib/python3.11/unittest/mock.py", line 950, in assert_called_once_with
    raise AssertionError(msg)
AssertionError: Expected 'get' to be called once. Called 2 times.
Calls: [call('https://example.com'), call('https://example.com')].

----------------------------------------------------------------------
Ran 752 tests in 81.140s

FAILED (failures=1, skipped=3)

> stop-emulator
> curl -X POST 'http://localhost:15606/shutdown'; pkill -f "CloudDatastore.jar"

Shutting down...
pkill: killing pid 99645 failed: Operation not permitted
Terminated
```
I've pushed the latest code for your ref.

**Comment by r3yc0n1c** [comment]
Hi @DanielRyanSmith could you please review these changes?

Thanks!

**Review by DanielRyanSmith — APPROVED** [review]
Thanks for the fix!

**Comment by r3yc0n1c** [comment]
> Thanks for the fix!

Thanks for merging 🙌! I want to continue contributing to Chromestatus for GSoC. Can you suggest more issues for me to work on?

**Comment by r3yc0n1c** [comment]
Also, I think GitHub might not be the most suitable platform for small talk or quick discussions. Do you have another preferred method of communication where we can chat more easily? Alternatively, if email works better for you, we can continue our discussions via raja.majumdar.01@gmail.com.

Thanks!

**Comment by DanielRyanSmith** [comment]
Hello @r3yc0n1c - feel free to send me an email with questions you'd like to discuss. You should be able to find my contact email through the GSoC project definitions.

**Pull request merged**